### DISCUSSION TITLE
Will all CSS variables be defined in SSR?

### DISCUSSION THREAD

**Discussion by AkiraVoid** [microsoft/fluentui#25714] [Q&A] [answered]
My site has many predefined CSS styles in document head while in SSR, but all variables are undefined.

**Comment by ling1726** [comment, marked as answer]
Hi, have you followed our SSR documentation?

https://react.fluentui.dev/?path=/docs/concepts-developer-server-side-rendering-basic-setup--page
https://react.fluentui.dev/?path=/docs/concepts-developer-server-side-rendering-nextjs-setup--page

**Reply by AkiraVoid** [reply]
Basically I have, I am not sure because I wrapped `FluentProvider` with my custom `ThemeProvider` which is used to switch between themes.

**Reply by ling1726** [reply]
Would you mind sharing a repro in the form of a sandbox or a repo? it's hard to make assumptions without more details at this point

**Reply by AkiraVoid** [reply]
OK, but maybe tomorrow, there's 00:30 am in my time zone.

**Reply by AkiraVoid** [reply]
@ling1726 https://github.com/AkiraVoid/fui-discussion-25714 After you start the server, you will see at first frame all styles are not applied, and then it get normal, and if you put the generated code in SSR into a normal html file, you will see all styles are inserted to document correctly but CSS variables are not loaded.

**Reply by AkiraVoid** [reply]
Sir, is it possible to reproduce this with my repo?

**Reply by ling1726** [reply]
Sorry for the wait, I just ran your repo and I see that the CSS variables are defined... maybe I've understood your original question incorrectly, could you please clarify?

![image](https://user-images.githubusercontent.com/20744592/203270869-dbd76d76-ff8b-464c-9868-8b4a3a3d0c7f.png)

**Reply by AkiraVoid** [reply]
Yes， I mean it seems like it will only load those variables on client because if I use the code generated by the server directly there are no variables are defined, which causes the first few frames of my app are not styled.

**Reply by layershifter** [reply]
@AkiraVoid There is a feature request to support it, https://github.com/microsoft/fluentui/issues/25425.
